### DISCUSSION TITLE
Any plans to update the dependencies of FakeXrmEasy?

### DISCUSSION THREAD

**Discussion by gvc55** [jordimontana82/fake-xrm-easy#601] [General]
I just noticed that the latest version of FakeXrmEasy cannot handle the latest FakeItEasy library (v7.3.1) stopping at v7.0. Any plans of updating these dependencies? Thanks in advance.

**Comment by BetimBeja** [comment]
Hi @gvc55,
did you read the notice on the readme?
FakeXrmEasy v2 and v3 are pretty much updated 😄 
https://github.com/jordimontana82/fake-xrm-easy#update-fakexrmeasy-v1x-is-deprecated-and-will-stop-receiving-updates-soon-start-planning-your-transition-to-versions-2x-and-3x-now

**Comment by jordimontana82** [comment]
Thank you for comment @gvc55   , Betim's response is correct, the latest versions already support FakeItEasy v7, they also support .net core and the license model has also changed. There is a new docs site with plenty of info to help you get started:

https://dynamicsvalue.github.io/fake-xrm-easy-docs/quickstart/

FXE v1 won't receive any major updates and it will be archived pretty soon...

**Comment by gvc55** [comment]
For the quick reply and Yes, I saw the readme. However, I am testing plug-ins which, as I understand, requires to be compiled using the .Net Framework and the v2 & v3 appear to be for .Net core. Am I missing something?Sent from my iPhoneOn Dec 20, 2022, at 6:45 AM, Betim Beja ***@***.***> wrote:﻿
Hi @gvc55,
did you read the notice on the readme?
FakeXrmEasy v2 and v3 are pretty much updated 😄
https://github.com/jordimontana82/fake-xrm-easy#update-fakexrmeasy-v1x-is-deprecated-and-will-stop-receiving-updates-soon-start-planning-your-transition-to-versions-2x-and-3x-now

**Reply by BetimBeja** [reply]
v2 is Full .Net Framework and v3 is .net core

**Reply by jordimontana82** [reply]
@gvc55  It's mentioned in these links, I'll probably add another mention on the home page so it's a bit more clear too 

https://dynamicsvalue.github.io/fake-xrm-easy-docs/quickstart/migrating-from-1x/
https://dynamicsvalue.github.io/fake-xrm-easy-docs/quickstart/installing/

**Comment by gvc55** [comment]
Thanks again for the reply, but where is V2? All I could see was .V9, which I understand is .Net core and .9 which is .Net Framework but it is not compatible with the latest FakeItEasy library. Sent from my iPhoneOn Dec 20, 2022, at 1:43 PM, Betim Beja ***@***.***> wrote:﻿
v2 is Full .Net Framework and v3 is .net core

**Reply by jordimontana82** [reply]
It's on the Installing guide I posted above....

https://dynamicsvalue.github.io/fake-xrm-easy-docs/quickstart/installing/


Install-Package FakeXrmEasy.v9 -Version 2.2.0

https://www.nuget.org/packages/FakeXrmEasy.v9/2.2.0

**Reply by jordimontana82** [reply]
The version is restricted to FakeItEasy v6 though now that I'm checking.

What functionality  do you need in FakeItEasy 7.3 that don't exist in 6.x?

**Comment by gvc55** [comment]
Thanks, now I understand. When I see V9 I do not expect a subversion but now I get it. In terms of what functionality I am expecting from FakeItEasy 7 vs 6, not really sure; I just went for the latest and then got the error that FakeXrm wasn’t up for it. Thanks again for the clarification. I will install V9 2.x and go from there.Sent from my iPhoneOn Dec 20, 2022, at 9:34 PM, Jordi ***@***.***> wrote:﻿
The version is restricted to FakeItEasy v6 though now that I'm checking.
What functionality  do you need in FakeItEasy 7.3 that don't exist in 6.x?

**Reply by jordimontana82** [reply]
No worries, yeah the .v9 is actually part of the package name, not the package version, and it's because we also support other target Crm/Dataverse instances (i.e messages and functionality exist in v9 that doesn't exist in say 8.2 on-prem, but plugins should really use that major version so it's a way of simplifying the installing and managing of these too).

The v9 used to be the .9 bit in the package name in FXE v1, but we changed it due to several reasons, one of them being NuGet package statistics having to be updated manually for every release....

**Reply by jordimontana82** [reply]
@gvc55  I would highly recommend going through the 6-step migration guide in PDF you can find at the bottom of this link first: https://dynamicsvalue.github.io/fake-xrm-easy-docs/quickstart/migrating-from-1x/

**Reply by jordimontana82** [reply]
What do you mean they did not run, that they aren't discovered or that they failed when running?

Without an actual exception is going to be difficult to troubleshoot, but I've seen at least one issue with that code: try not using static variables, as they might be shared across unit tests and multiple threads.

Did you install the FakeXrmEasy.CodeActivities package?

What package dependencies do you have installed? Can you share these?

**Comment by gvc55** [comment]
Sorry, but I’ve been under so much pressure that I couldn’t see the forest from the trees. It was my bad all the way in that the I had change the signature of the main methods, classes (initialize, cleanup, etc) but never saw the error until I picked the Test tab on the output window. It needed to be static and I removed that thinking it has changed. Oh well, it all good. Sorry you saw the post because I deleted it but not in time, I guess. 😟Sent from my iPhoneOn Dec 21, 2022, at 3:52 PM, Jordi ***@***.***> wrote:﻿
What do you mean they did not run, that they aren't discovered or that they failed when running?
Without an actual exception is going to be difficult to troubleshoot, but I've seen at least one issue with that code: try not using static variables, as they might be shared across unit tests and multiple threads.
Did you install the FakeXrmEasy.CodeActivities package?
What package dependencies do you have installed? Can you share these?

**Reply by jordimontana82** [reply]
No problem, all good now , so? @gvc55

**Comment by gvc55** [comment]
Yup, so far so good. I am taking a few days off for the holidays so will have a better after the new year. But yes, I expect that everything will be just fine. Thanks so much for the  support.Sent from my iPhoneOn Dec 22, 2022, at 3:00 AM, Jordi ***@***.***> wrote:﻿
No problem, all good now , so? @gvc55
